clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerMembershipManagerTest.java

```diff
@@ -1617,10 +1617,25 @@ public void testRevokePartitionsUsesTopicNamesLocalCacheWhenMetadataNotAvailable
     }
 
     @Test
-    public void testOnSubscriptionUpdatedTransitionsToJoiningOnlyIfNotInGroup() {
+    public void testOnSubscriptionUpdatedDoesNotSetShouldTransitionToJoiningIfInGroup() {
         ConsumerMembershipManager membershipManager = createMemberInStableState();
         membershipManager.onSubscriptionUpdated();
+        assertTrue(membershipManager.hasSubscriptionUpdated());
+        membershipManager.maybeJoinGroup();
         verify(membershipManager, never()).transitionToJoining();
+        assertFalse(membershipManager.hasSubscriptionUpdated());
+    }
+
+    @Test
+    public void testOnSubscriptionUpdatedSetShouldTransitionToJoiningIfNotInGroup() {
```

Comment:
`testOnSubscriptionUpdatedTransitionsToJoiningIOnPollfNotInGroup`?

Reply:
Changed the test case name to `testOnSubscriptionUpdatedTransitionsToJoiningOnPollIfNotInGroup`. Thanks.